rmf_task_ros2/include/rmf_task_ros2/dispatcher/Dispatcher.hpp

```diff
@@ -0,0 +1,97 @@
+/*
+ * Copyright (C) 2020 Open Source Robotics Foundation
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+*/
+
+#ifndef RMF_TASK_ROS2__DISPATCHER__NODE_HPP
+#define RMF_TASK_ROS2__DISPATCHER__NODE_HPP
+
+#include <rclcpp/node.hpp>
+#include <rmf_utils/impl_ptr.hpp>
+#include <rmf_utils/optional.hpp>
+
+#include <rmf_task_ros2/StandardNames.hpp>
+#include <rmf_task_ros2/bidding/Auctioneer.hpp>
+#include <rmf_task_ros2/action/ActionInterface.hpp>
+
+namespace rmf_task_ros2 {
+namespace dispatcher {
+
+//==============================================================================
+using DispatchTasks = std::map<TaskID, TaskStatusPtr>;
+using DispatchTasksPtr = std::shared_ptr<DispatchTasks>;
+using DispatchState = TaskStatus::State;
+
+//==============================================================================
+
+class Dispatcher : public std::enable_shared_from_this<Dispatcher>
```

Comment:
As mentioned during our meeting, I believe a cleaner way to implement this would be to have a pure abstract interface for the middleware itself, which will be taken in by this dispatcher node. This way we can implement the POST/GET middleware on its own, which the dispatcher node will use.

Reply:
For middleware-agnostic interfaces, I would recommend putting the interface outside of any packages ending in `_ros2` and then only putting the ROS2-specific implementation of the interface into the `_ros2` package.